core/domain/user_services.py

```diff
@@ -105,6 +105,7 @@ def __init__(
 
         Args:
             user_id: str. The unique ID of the user.
+            gae_user_id: str. The ID of the user retrieved from GAE.
```

Comment:
As I'm reading this PR, I keep having to think twice when reading "gae_user_id", and I am a bit worried that developers may get confused and mix up "gae_user_id" with "user_id", which could be problematic.

Could we use user.gae_id instead? It might be too late for the storage layer if the migration is already done (is it?), but in that case we could at least use that in the domain layer and above. What do you think?

Reply:
`UserGaeIdOneOffJob` wasn't yet run on production, so we might scratch it and modify for the next release. cc @nithusha21 @ankita240796

Reply:
Noted, Thanks!

Reply:
> Could we use user.gae_id instead?

@seanlip, do you wanted to suggest `user_gae_id` here?